### PR TITLE
Check source data change before full refresh

### LINKED ISSUE #190: Full refresh should check -no data change- and abort instead of doing full refresh
I think it would be good to check the condition - no data change - for "full" refresh mode?
It doesn't need to be handled with this PR, but it seems necessary.

_Originally posted by @sezruby in https://github.com/microsoft/hyperspace/pull/189#issuecomment-705325520_

### PR DESCRIPTION
**Pull request by sezruby**
### What is the context for this pull request?

 - **Tracking Issue**: Fixes #190
 - **Parent Issue**: n/a
 - **Dependencies**: n/a

### What changes were proposed in this pull request?

Check source data change before full refresh and abort if there is no change.

### Does this PR introduce _any_ user-facing change?

Yes, no version update (and unnecessary data shuffle) when full refresh without any source data change.

### How was this patch tested?

Unit test

### CODE REVIEW

File: src/main/scala/com/microsoft/hyperspace/actions/RefreshAction.scala [hunk 1]
```diff
@@ -40,6 +40,20 @@ class RefreshAction(
...
+  /**
+   * Validate index is in active state for refreshing and there are some changes in
+   * source data file(s).
+   */
+  final override def validate(): Unit = {
+    super.validate()
+
+    if (currentFiles.equals(previousIndexLogEntry.allSourceFileInfos)) {
+      // scalastyle:off
+      println("here")
```
Comment (pirz) [inline src/main/scala/com/microsoft/hyperspace/actions/RefreshAction.scala:52]: I guess you had added below these two lines for quick debugging and you may wanna delete them now

// scalastyle:off
   println("here")

File: src/test/scala/com/microsoft/hyperspace/actions/RefreshActionTest.scala [hunk 2]
```diff
@@ -83,16 +94,26 @@ class RefreshActionTest extends SparkFunSuite with SparkInvolvedSuite {
...
     val ex = intercept[HyperspaceException](action.validate())
     assert(ex.getMessage.contains("Refresh is only supported in ACTIVE state"))
   }
+
+  test("validate() fails if there is no source data change.") {
+    testLogEntry.state = ACTIVE
+    when(mockLogManager.getLog(anyInt)).thenReturn(Some(testLogEntry))
+    val action = new RefreshAction(spark, mockLogManager, mockDataManager)
+    val ex = intercept[NoChangesException](action.validate())
+    assert(ex.getMessage.contains("Refresh full aborted"))
```
Comment (pirz) [inline src/test/scala/com/microsoft/hyperspace/actions/RefreshActionTest.scala:117]: What about checking the full exception msg as it is a fixed/non-parametrized one?

`assert(ex.getMessage.contains("Refresh full aborted as no source data changed"))`

**Review by pirz — COMMENTED** [review]
LGTM, thanks @sezruby

**Review by imback82 — APPROVED** [review]
LGTM, thanks @sezruby!

**Pull request merged**